jbang export does not handle it well when asked to put into directory that does not exist

```
 jbang export fatjar -O target/x.jar src/main/java/org/jreleaser/examples/HelloWorld.java
[jbang] [ERROR] /Users/manderse/code/maxandersen/helloworld-java-bin/target/x.jar
[jbang] Run with --verbose for more details
```

no info about what the error is.

`--verbose` gives a little hint

```
jbang --verbose export fatjar -O target/x.jar src/main/java/org/jreleaser/examples/HelloWorld.java
[jbang] [0:422] jbang version 0.106.3
[jbang] [0:435] Resolving resource ref: src/main/java/org/jreleaser/examples/HelloWorld.java
[jbang] [0:438] Resolved resource ref as: src/main/java/org/jreleaser/examples/HelloWorld.java (cached as: /Users/manderse/code/maxandersen/helloworld-java-bin/src/main/java/org/jreleaser/examples/HelloWorld.java)
[jbang] [0:483] Looking for JDK: 0
[jbang] [0:488] Using JDK: 17 (17.0.5, current, /Users/manderse/.sdkman/candidates/java/22.3.r17-grl)
[jbang] [0:489] No build required. Reusing jar from /Users/manderse/.jbang/cache/jars/HelloWorld.java.77d5522647460211974d38c0166fe9dfa5c00558c2ec1df47f2d4ec07277992a/HelloWorld.jar
[jbang] [0:490] [ERROR] /Users/manderse/code/maxandersen/helloworld-java-bin/target/x.jar
java.nio.file.NoSuchFileException: /Users/manderse/code/maxandersen/helloworld-java-bin/target/x.jar
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixCopyFile.copyFile(UnixCopyFile.java:246)
	at java.base/sun.nio.fs.UnixCopyFile.copy(UnixCopyFile.java:603)
	at java.base/sun.nio.fs.UnixFileSystemProvider.copy(UnixFileSystemProvider.java:257)
	at java.base/java.nio.file.Files.copy(Files.java:1305)
	at dev.jbang.cli.ExportFatjar.apply(Export.java:367)
	at dev.jbang.cli.BaseExportCommand.doCall(Export.java:75)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:145)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:21)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at dev.jbang.cli.JBang$3.handle(JBang.java:147)
	at dev.jbang.cli.JBang$3.handle(JBang.java:142)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at dev.jbang.Main.main(Main.java:14)
[jbang] [0:490] If you believe this a bug in jbang, open an issue at https://github.com/jbangdev/jbang/issues
```

but should not have to run with verbose to get that error + should have option to just create that directory. I would argue it should just do so and only fail if existing file there which then can be `--force` if need be.

--- a/src/main/java/dev/jbang/cli/Export.java
+++ b/src/main/java/dev/jbang/cli/Export.java
@@ -114,11 +114,13 @@ class ExportLocal extends BaseExportCommand {
 		Path outputPath = getJarOutputPath();
 		if (outputPath.toFile().exists()) {
 			if (exportMixin.force) {
-				outputPath.toFile().delete();
+				Util.deletePath(outputPath, false);
 			} else {
 				Util.errorMsg("Cannot export as " + outputPath + " already exists. Use --force to overwrite.");
 				return EXIT_INVALID_INPUT;
 			}
+		} else {
+			Util.mkdirs(outputPath.getParent());
 		}
 		Files.copy(source, outputPath);
 
@@ -150,11 +152,13 @@ class ExportPortable extends BaseExportCommand {
 		Path outputPath = getJarOutputPath();
 		if (outputPath.toFile().exists()) {
 			if (exportMixin.force) {
-				outputPath.toFile().delete();
+				Util.deletePath(outputPath, false);
 			} else {
 				Util.errorMsg("Cannot export as " + outputPath + " already exists. Use --force to overwrite.");
 				return EXIT_INVALID_INPUT;
 			}
+		} else {
+			Util.mkdirs(outputPath.getParent());
 		}
 
 		Files.copy(source, outputPath);
@@ -208,7 +212,7 @@ class ExportMavenPublish extends BaseExportCommand {
 				return EXIT_INVALID_INPUT;
 			}
 			if (exportMixin.force) {
-				outputPath.toFile().mkdirs();
+				Util.mkdirs(outputPath);
 			} else {
 				Util.errorMsg("Cannot export as " + outputPath + " does not exist. Use --force to create.");
 				return EXIT_INVALID_INPUT;
@@ -301,11 +305,13 @@ class ExportNative extends BaseExportCommand {
 		Path outputPath = getNativeOutputPath();
 		if (outputPath.toFile().exists()) {
 			if (exportMixin.force) {
-				outputPath.toFile().delete();
+				Util.deletePath(outputPath, false);
 			} else {
 				Util.errorMsg("Cannot export as " + outputPath + " already exists. Use --force to overwrite.");
 				return EXIT_INVALID_INPUT;
 			}
+		} else {
+			Util.mkdirs(outputPath.getParent());
 		}
 		Files.copy(source, outputPath);
 
@@ -340,11 +346,13 @@ class ExportFatjar extends BaseExportCommand {
 		Path outputPath = getFatjarOutputPath();
 		if (outputPath.toFile().exists()) {
 			if (exportMixin.force) {
-				outputPath.toFile().delete();
+				Util.deletePath(outputPath, false);
 			} else {
 				Util.errorMsg("Cannot export as " + outputPath + " already exists. Use --force to overwrite.");
 				return EXIT_INVALID_INPUT;
 			}
+		} else {
+			Util.mkdirs(outputPath.getParent());
 		}
 
 		List<ArtifactInfo> deps = prj.resolveClassPath().getArtifacts();

--- a/src/test/java/dev/jbang/cli/TestExport.java
+++ b/src/test/java/dev/jbang/cli/TestExport.java
@@ -32,7 +32,7 @@ public class TestExport extends BaseTest {
 	@Test
 	void testExportFile() throws Exception {
 		String src = examplesTestFolder.resolve("helloworld.java").toString();
-		String outFile = cwdDir.resolve("helloworld.jar").toString();
+		String outFile = cwdDir.resolve("subdir/helloworld.jar").toString();
 		CaptureResult result = checkedRun(null, "export", "local", "-O", outFile, src);
 		assertThat(result.err, matchesPattern("(?s).*Exported to.*helloworld.jar.*"));
 		assertThat(new File(outFile), anExistingFile());
@@ -186,5 +186,14 @@ public class TestExport extends BaseTest {
 			ExportLocal export = (ExportLocal) pr.subcommand().subcommand().commandSpec().userObject();
 			export.doCall();
 		});
+	}
+
+	@Test
+	void testExportFatJar() throws Exception {
+		String src = examplesTestFolder.resolve("helloworld.java").toString();
+		String outFile = cwdDir.resolve("subdir/helloworld.jar").toString();
+		CaptureResult result = checkedRun(null, "export", "fatjar", "-O", outFile, src);
+		assertThat(result.err, matchesPattern("(?s).*Exported to.*helloworld.jar.*"));
+		assertThat(new File(outFile), anExistingFile());
 	}
 }

